unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1823,76 +1849,126 @@ protected bool checkForUpDownAngleLimit(string direction, float degrees)
         }
 
         //teleport full, base version does not consider being able to hold objects
-        public virtual void TeleportFull(ServerAction action) {
-            targetTeleport = new Vector3(action.x, action.y, action.z);
+        public virtual void TeleportFull(
+            float x,
+            float y,
+            float z,
+            Vector3 rotation,
```

Comment:
One other request, both for Teleport's new `rotation` and TeleportFull's existing rotation. Can we overload this from the C# side such that rotation can also take in a float on the BaseFPS and PhysicsFPS?

This is already happen in an extremely dangerous fashion from the Python side: https://github.com/allenai/ai2thor/blob/master/ai2thor/controller.py#L840 and it is quite more easier to now have to type
```python
rotation=myActualRotation
```
instead of
```python
rotation=dict(x=0, y=<myActualRotation>, z=0)
```

Reply:
I tried this override and it breaks the action dispatcher unfortunately :(

Reply:
It will try to do things like type casting a dictionary as a float and fail obscurely.

Reply:
Did you try creating a method like this:

```
        public void TeleportFull(
            float x,
            float y,
            float z,
            float rotation) {
   TelportFull(x, y, z, new Vector3(0.0f, rotation, 0.0f));
}
```

That should work - let me know if it does not.

The only issue I have with assuming rotation always equals rotation about the y axis is that it presumes the type of agent and how it can rotate.  A drone for example could rotate in all dimensions.

Reply:
Maybe I misunderstood the error I was getting but when I tried adding a method with arguments:
```C#
public void TeleportFull(
            float x,
            float y,
            float z,
            float rotation,
            float horizon,
            bool standing,
            bool forceAction = false
) { ... }
```
it failed and complained about being unable to type cast rotation appropriately.

Reply:
What were you passing from the Python side to the step() function?

Reply:
@ekolve  Oh, are we ok with removing the line that @mattdeitke pointed out? I'm not sure what the impact of this would be for existing code.

Reply:
Preferably don't to remove it until `TeleportFull` works with `rotation: float` for the default and locobot agents. These change in bodily rotations with respect to their yaw. I use it in a bunch of places with teleport.